sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -1899,6 +1899,21 @@ class DatasetSuite extends QueryTest
     val e = intercept[AnalysisException](spark.range(1).tail(-1))
     e.getMessage.contains("tail expression must be equal to or greater than 0")
   }
+
+  test("sameSemantics and semanticHash work") {
```

Comment:
Shall we add `SPARK-30791:` prefix although it is a feature API per "Pull Request" at https://spark.apache.org/contributing.html